### ISSUE TITLE
Multiple installation issues with PHP_CS_Fixer in Atom

### ISSUE THREAD

**Issue by mjoycemilburn** [FriendsOfPHP/PHP-CS-Fixer#5200]
Hi. I'm new to Atom and PHP_CS_Fixer, so please bear with me. I really appreciate the work you're doing here and seem to have got things working now, but it was a struggle. Google searches around the issues I've encountered suggest that others seem to have encountered difficulties too. I'm posting this for three reasons - (a) in the hope that it may help others, (b) to request confirmation that the fixes I've applied are appropriate. and (c) to suggest that greater clarity in PHP_CS_Fixer's README and the PHP_CS_Fixer setting page in Atom would be welcome.

After installing Atom, PHP, Atom-Beautify and PHP_CS_Fixer, I was rather upset to find that Fixer was having absolutely no effect on my php files. On save, a squirrly "Beautification in progress" popup appeared, but even the simplest indentation or syntax error in my files was left unchanged.

Going back to the Readme for PHP_CS_Fixer I followed the instructions for running the plugin in Powershell and was relieved to find that it worked just fine when pointed directly at my file as per:

$ php php-cs-fixer.phar fix /path/to/file

So the problem seemed to be the Atom settings for PHP_CS_Fixer:

PHP_CS_Fixer executable path
PHP_CS_Fixer Rules
PHP_CS_Fixer Config file path

It took me a while to work out that "executable path" referred to the phar file - helpful error messages appeared when I left this blank and tried to "beautify". "Fixer rules" remains a mystery, so I left this blank. PHP_CS_Fixer Config file path completely stumped me as I had no idea what a config file might look like, whether or not I needed it at all or, if I did, where I should put it. As far as I can remember now, I had the location set to the same location as my phar file (which I'd placed in C:\Users\me\AppData\Local\atom\php-cs-fixer) but I hadn't placed a file there.

It took me rather longer to realise that the Packages/Atom Beautify/Debug command was the way to find out what was happening. Part of the reason was that I was working, initially, on very large files. The information I needed lay at the bottom and I wasn't aware that it was even there till I stumbled upon it by chance. There's not much documentation on this tool and more would have been very helpful.  

Anyway, I could now see that there was something wrong because two messages reported that TEMP files that the system had created (I guessed these were copies of my original editor source before and after processing) could not be read.

Working on the assumption that I needed to do something with my config file, I took the example from Readme shown below and placed it in C:\Users\me - ie the same folder that contained my .atom installation:

<?php

$finder = PhpCsFixer\Finder::create()
    ->exclude('somedir')
    ->notPath('src/Symfony/Component/Translation/Tests/fixtures/resources.php')
    ->in(__DIR__)
;

return PhpCsFixer\Config::create()
    ->setRules([
        '@PSR2' => true,
        'strict_param' => true,
        'array_syntax' => ['syntax' => 'short'],
    ])
    ->setFinder($finder)
;

I know this example is intended to guide experts tweaking their rules, but since its coding employs some rather advanced PHP syntax I didn't dare amend it - as far as I could see it wasn't going to take me far off the straight and narrow.

After a few false starts (it wasn't immediately clear that placing this in C:\Users\me was the right thing to do - the succinct instruction on the Atom PHP_CS_Fixer page referred to "the root path of the current project" - what project?) corrected by helpful error messages in the Debug report and regular Atom restarts (and even a reboot at one point, I recall) PHP_CS_Fixer started to work. Yay!

Goodness though, that was hard work.

In conclusion, could I request some clear advice on whether or not a config file /does/ need to be manually installed for Atom use and, if so, where this should be placed and what the default content should be? I note, now that there's a .php_cs.dist file in the repo but the content of this looks even scarier than the file I'm using and, again, there are few clues as to what exactly it's doing.

I'm sure the instructions in README are fine for experts but, given that Atom has a lot to offer PHP learners and that the facilities in PHP_CS_Fixer are so useful, it would be good to see something a bit more expansive in this area.

Keep up the good work!

**Comment by SpacePossum** [comment]
Hi @mjoycemilburn and thanks for sharing your experience with the tool with us.

I take it from your story it wasn't an easy start, thanks for keep trying and I hope you got it running :)

We take care to provide a way for tooling like Atom and PHPStorm and others like CI pipeline workers to work or integrate with the tool. However, we are typically not focused on the integration itself here. Sure, some of us use those toolings, but these aren't developed here.
I saw you also shared some ideas and details with the Atom team, sounds like a good idea :)

Maybe we can help you a bit running the tool without Atom, but directly from a terminal? The README can be a bit intimidating at the start, but if you setup a safe playground you can have a go without to much risk.
I would recommend taking your project and make sure you have all your changes stashed or committed or stored so if the tool make changes you don't like you can revert back.
And than giving it a go, so download the phar, following the docs here (if you haven't already)

https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/doc/installation.rst#globally-manual

Follow the instructions till `Then, just run php-cs-fixer.`  (this will work for Mac and Linux)

If this works I recommend running this:
>  php-cs-fixer fix src --dry-run -vvv

if your code is within the folder `src`

Now from we can start making a configuration file, but to help you do that I would need to know some more details about your project.

Lets try above first, if you have any more questions after that feel free to visit us here:
https://gitter.im/PHP-CS-Fixer/Lobby 

it is a free chat place that you can enter with your github account to talk, which is more effective than trying here.
Good luck!
